core/src/scene/sceneLoader.cpp

```diff
@@ -177,470 +141,323 @@ void SceneLoader::applyGlobals(Node root, Scene& scene) {
-            LOGNode("Mixing styles: '%s'", styles, e.what());
+            loadCameras(cameras, _sceneCamera);
         }
-        for (const auto& entry : styles) {
-            try {
-                auto name = entry.first.Scalar();
-                auto config = entry.second;
-                loadStyle(_platform, name, config, _scene);
-            }
-            catch (YAML::RepresentationException e) {
-                LOGNode("Parsing style: '%s'", entry, e.what());
-            }
+        catch (const YAML::RepresentationException& e) {
+            LOGNode("Parsing cameras: '%s'", cameras, e.what());
         }
     }
+}
 
-    // Styles that are opaque must be ordered first in the scene so that
-    // they are rendered 'under' styles that require blending
-    std::sort(_scene->styles().begin(), _scene->styles().end(), Style::compare);
+void SceneLoader::loadCameras(const Node& _cameras, SceneCamera& _sceneCamera) {
+    // To correctly match the behavior of the webGL library we'll need a place
+    // to store multiple view instances.  Since we only have one global view
+    // right now, we'll just apply the settings from the first active camera we
+    // find.
+
+    for (const auto& entry : _cameras) {
+        loadCamera(entry.second, _sceneCamera);
+    }
+}
 
-    // Post style sorting set their respective IDs=>vector indices
-    // These indices are used for style geometry lookup in tiles
-    auto& styles = _scene->styles();
-    for(uint32_t i = 0; i < styles.size(); i++) {
-        styles[i]->setID(i);
+void SceneLoader::loadCamera(const Node& _camera, SceneCamera& _sceneCamera) {
+    auto& camera = _sceneCamera;
 
-        if (auto pointStyle = dynamic_cast<PointStyle*>(styles[i].get())) {
-            pointStyle->setTextures(_scene->textures());
+    if (const Node& active = _camera["active"]) {
+        if (!YamlUtil::getBoolOrDefault(active, false)) {
+            return;
         }
     }
 
-    if (Node layers = config["layers"]) {
-        for (const auto& layer : layers) {
-            try { loadLayer(layer, _scene); }
-            catch (YAML::RepresentationException e) {
-                LOGNode("Parsing layer: '%s'", layer, e.what());
+    auto type = _camera["type"].Scalar();
+    if (type == "perspective") {
+        camera.type = CameraType::perspective;
+
+        // Only one of focal length and FOV is applied;
+        // according to docs, focal length takes precedence.
+        if (const Node& focal = _camera["focal_length"]) {
+            float floatValue;
+            if (YamlUtil::getFloat(focal, floatValue)) {
+                camera.fieldOfView = View::focalLengthToFieldOfView(floatValue);
+            } else if (focal.IsSequence()) {
+                camera.fovStops = std::make_shared<Stops>(Stops::Numbers(focal));
+                for (auto& f : camera.fovStops->frames) {
+                    f.value = View::focalLengthToFieldOfView(f.value.get<float>());
```

Comment:
Prefer `YamlUtil::getFloatOrDefault` over `get<float>` - if we remove all instances of `get<T>` I think we can remove the `try {} catch () {}` around the camera loaders.

Reply:
It's a `Frame::StopValue`, not a yaml node